graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetConfig.java

```diff
@@ -103,35 +100,21 @@
     @NotNull
     public abstract ZonedDateTime creationDate();
 
-    // TODO 2.2: Migrate config setting to index set!
     @JsonProperty("index_analyzer")
-    @JsonIgnore
-    public String indexAnalyzer() {
-        return "standard";
-    }
+    @NotBlank
+    public abstract String indexAnalyzer();
 
-    // TODO 2.2: Migrate config setting to index set!
     @JsonProperty("index_template_name")
-    @JsonIgnore
-    public String indexTemplateName() {
-        return "graylog-internal-" + requireNonNull(id());
```

Comment:
I think we should handle all defaults in the `@JsonCreator` method in this class because that's the place where the fields can be `null` when coming out of the database. We should remove the `@Nullable` annotation from the new field methods because they shouldn't ever be null. Otherwise we have to adjust all consumers of the methods to handle the `null` case. (they don't do that now)